Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -66,42 +47,80 @@ private extension ClangModule {
     }
 }
 
-private extension Sources {
-    func compilePathsForBuildDir(_ wd: String) -> [(filename: String, source: String, object: String, deps: String)] {
-        return relativePaths.map { source in
-            let path = Path.join(root, source)
-            let object = Path.join(wd, "\(source).o")
-            let deps = Path.join(wd, "\(source).d")
+/// A helper struct for ClangModule to compute basic
+/// flags needed to compile and link C language targets.
+struct ClangModuleBuildMeta {
```

Comment:
Why is this better than having the methods on the ClangModule?

I agree we need improvements in this direction, but I'm not completely sure this is the best approach.